Java/Mazes/src/DrawMaze.java

```diff
@@ -0,0 +1,427 @@
+import java.awt.BasicStroke;
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Random;
+import java.util.Set;
+import java.util.Stack;
+
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+
+public class DrawMaze extends JPanel {
+
+    /**
+     * The width of the maze halls, a constant
+     */
+    private static final int SIZE = 15;
+    /**
+     * The thickness of the maze walls, a constant
+     */
+    private static final int WALL = 2;
+    /**
+     * The thickness of the border around the maze, a constant
+     */
+    private static final int BORDER = 5;
+    /**
+     * The width of the maze given in the constructor
+     */
+    private final int mazeWidth;
+    /**
+     * The height of the maze given in the constructor
+     */
+    private final int mazeHeight;
+    // storage for the walls that are added
+    private final Set<Wall> walls;
+
+    /**
+     * Construct a simple maze
+     *
+     * @param width  The width of the maze in "cells"
+     * @param height The height of the maze in "cells"
+     */
+    private DrawMaze(int width, int height) {
+        mazeWidth = width;
+        mazeHeight = height;
+        setPreferredSize(new Dimension(width * SIZE + 2 * BORDER + WALL,
+                height * SIZE + 2 * BORDER + WALL));
+        walls = new HashSet<>();
+    }
+
+    /**
+     * Sample application method, showing how to use DrawMaze
+     *
+     * @param args The command-line arguments
+     */
+    public static void main(String[] args) {
+        DrawMaze myMaze = new DrawMaze(75, 60);
+        DrawMaze myMaze1 = new DrawMaze(75, 60);
+
+        // myMaze.maze1();
+        myMaze1.maze2();
+    }
+
+    /**
+     * Add a vertical wall to the maze
+     *
+     * @param x   The horizontal offset for the wall
+     * @param y   The vertical offset for the wall
+     * @param len The length of the wall
+     * @return true, if the requested wall added to the set of walls
+     */
+    private boolean addVerticalWall(int x, int y, int len) {
+        if (x > mazeWidth || y + len > mazeHeight) throw new IllegalArgumentException("Wall exceeds maze boundary");
+        boolean added = false;
+        for (int i = 0; i < len; i++) {
```

Comment:
What are you iterating over?  I know it is the length of the wall, but, is it pixles, or something else?

Reply:
This was originally going to be used for error checking in one maze generating algorithm but then I ended up not calling it. It involved varying length walls and confirming that the length of wall was valid. I have removed it now and will reimplement if the times comes that another algorithm has use of it.